src/CommunityToolkit.Maui/Converters/MathExpressionConverter/MathExpression.shared.cs

```diff
@@ -127,15 +123,12 @@ public double Calculate()
 		return stack.Pop();
 	}
 
+	[GeneratedRegex(@"(?<!\d)\-?(?:\d+\.\d+|\d+)|\+|\-|\/|\*|\(|\)|\^|\%|\,|\w+")]
+	private static partial Regex ValidExpressionRegex();
```

Comment:
Maybe we should get a more meaningful name like MathExpressionRegexPattern?

Reply:
Agreed! To be honest, I named it this because I couldn't actually figure out what it does. (I don't really know much about reading/writing RegEx)

Reply:
```suggestion
	private static partial Regex MathExpressionRegexPattern();
```